vividus-to-xray-exporter/src/main/java/org/vividus/xray/converter/ManualStepConverter.java

```diff
@@ -0,0 +1,134 @@
+            .compile(String.format(OPTIONAL_PARTS_FORMAT, DATA_SIGN, DATA_SIGN, RESULT_SIGN), DOTALL_MULTILINE_MODE);
+    private static final Pattern RESULT_PATTERN = Pattern
+            .compile(String.format(OPTIONAL_PARTS_FORMAT, RESULT_SIGN, RESULT_SIGN, DATA_SIGN), DOTALL_MULTILINE_MODE);
+
+    public List<ManualTestStep> convert(String storyTitle, String scenarioTitle, List<Step> steps)
+    {
+        boolean checkPrefix = steps.stream()
+                                   .map(Step::getOutcome)
+                                   .allMatch("comment"::equals);
+
+        if (!checkPrefix)
+        {
+            throw new ManualStepConversionException(getErrorMessage(storyTitle, scenarioTitle));
+        }
+
+        String manualScenario = steps.stream().map(Step::getValue)
+                .map(v -> RegExUtils.replaceAll(v, "!--\\s*-?\\s*", StringUtils.EMPTY))
+                .collect(Collectors.joining(lineSeparator()));
+
+        if (!manualScenario.startsWith(STEP_SIGN))
+        {
+            throw new ManualStepConversionException(getErrorMessage(storyTitle, scenarioTitle));
+        }
+
+        List<ManualTestStep> manualSteps = new ArrayList<>();
+        Matcher stepMatcher = STEP_BLOCK_PATTERN.matcher(manualScenario);
+        while (stepMatcher.find())
+        {
+            String stepBlock = stepMatcher.group();
+            String step = extractFirstByPattern(STEP_PATTERN, stepBlock, STEP_SIGN);
+            ManualTestStep manualTestStep = new ManualTestStep(step);
+            extractFirstByPattern(DATA_PATTERN, stepBlock, DATA_SIGN, manualTestStep::setData);
+            extractFirstByPattern(RESULT_PATTERN, stepBlock, RESULT_SIGN, manualTestStep::setExpectedResult);
+            manualSteps.add(manualTestStep);
+        }
+        return manualSteps;
+    }
+
+    private static String extractFirstByPattern(Pattern pattern, String data, String target)
+    {
+        Matcher matcher = pattern.matcher(data);
+        if (matcher.find())
+        {
+            String matched = matcher.group(1);
+            if (matcher.find())
+            {
+                throw new ManualStepConversionException(
+                        String.format("Only one %s is expected to be present in the data", target));
+            }
+            return matched.strip();
+        }
+        return null;
+    }
+
+    private static void extractFirstByPattern(Pattern pattern, String data, String target,
+            Consumer<String> valueConsumer)
+    {
+        Optional.ofNullable(extractFirstByPattern(pattern, data, target)).ifPresent(valueConsumer);
+    }
+
+    private static String getErrorMessage(String storyTitle, String scenarioTitle)
+    {
+        return "Error:" + lineSeparator() + "Story: " + storyTitle + lineSeparator() + "Scenario: " + scenarioTitle
+                + lineSeparator() + FORMAT_MESSAGE;
+    }
+
+    public static class ManualStepConversionException extends RuntimeException
```

Comment:
why is `RuntimeException`?

Reply:
converted to exception due to new requirements regarding to errors tracking